### DISCUSSION TITLE
Add support for sending message to unsaved number

### DISCUSSION THREAD

**Discussion by Docbroke** [d99kris/nchat#303] [Ideas]
I have two related request here
1. Send message to unsaved number, like when we search chats/contacts we may use mo. number as filter and choose that to send message.
2.  Send PM to someone in group, by selecting his/her message and using assigned keyword

**Comment by d99kris** [comment]
Hi @Docbroke - taking note of the request(s).

**Comment by desgua** [comment]
I wonder if that would trigger a ban from WhatsApp automatic spam detection.

**Reply by d99kris** [reply]
Possibly, but I think an implementation would involve adding the recipient as a new contact (and thus a little bit safer w.r.t. spam), but I have not looked at this and whether official app allows sending without adding contact. 

**Edit:** Re-reading the title of the feature request I agree, it may be an issue for unsaved numbers, but group members are probably ok.

**Reply by Docbroke** [reply]
This is actually available officially. One can use `$BROWSER` https://web.whatsapp.com/send/?phone="$P.Number" with whatsapp web.
Some android contact apps also allows directly opening whatsapp message page for given contact number, (e.g,, open-contacts on f-droid)

**Reply by desgua** [reply]
My worry is if it would be easier for WhatsApp automatic ban system to detect an unofficial app sending an initial message because the first handshake requires more setting up (encryption keys for example).

**Reply by desgua** [reply]
*I'm not an expert, I'm just sharing my thoughts.

**Reply by d99kris** [reply]
> This is actually available officially. One can use `$BROWSER` [https://web.whatsapp.com/send/?phone="$P.Number](https://web.whatsapp.com/send/?phone=%22$P.Number)" with whatsapp web. Some android contact apps also allows directly opening whatsapp message page for given contact number, (e.g,, open-contacts on f-droid)

Thanks for pointing out, makes sense.

> My worry is if it would be easier for WhatsApp automatic ban system to detect an unofficial app sending an initial message because the first handshake requires more setting up (encryption keys for example).
> *I'm not an expert, I'm just sharing my thoughts.

No worries, thanks for sharing. So far I haven't seen WhatsApp bans based on poor endpoint interaction. From what I see online it's only poor service interaction (as in actually spamming non-contacts) that gets people banned. But yes, it's an area to be more careful with. Just as long as providing similar functionality as the official (web) app, it should be fairly safe.

**Comment by desgua** [comment]
My two cents on how to implement it can be seen [here](https://github.com/d99kris/nchat/compare/master...desgua:nchat:add_message_new_contact).
